indico/web/util.py

```diff
@@ -181,31 +185,84 @@ def url_for_index(_external=False, _anchor=None):
     return url_for('categories.display', _external=_external, _anchor=_anchor)
 
 
-def signed_url_for(user, blueprint, url_params=None, *args, **kwargs):
-    """Get a URL from a blueprint, which is signed using a user's signing secret."""
-    from indico.web.flask.util import url_for
-    _external = kwargs.pop('_external', False)
-    base_url = url_for(blueprint, *args, **(url_params or {}))
-    qs = url_encode(sorted(kwargs.items()))
-    # this is the URL which is to be signed
-    url = f'{base_url}?{qs}' if qs else base_url
+def is_legacy_signed_url_valid(user, url):
+    """Check whether a legacy signed URL is valid for a user.
+
+    This util is deprecated and only exists because people may be actively
+    using URLs using the old style token. Any new code should use the new
+    :func:`signed_url_for_user` and :func:`verify_signed_user_url` utils
+    which encode the user id within the signature.
+    """
+    parsed = url_parse(url)
+    params = url_decode(parsed.query)
+    try:
+        signature = params.pop('token')
+    except KeyError:
+        return False
+
+    url = url_unparse((
+        '',
+        '',
+        parsed.path,
+        url_encode(sorted(params.items()), sort=True),
+        parsed.fragment
+    ))
+    signer = Signer(user.signing_secret, salt='url-signing')
+    return signer.verify_signature(url.encode(), signature)
```

Comment:
Don't we need to encode the signature as before? (also applies to other occasions below)
```suggestion
    return signer.verify_signature(url.encode(), signature.encode())
```

Reply:
No, it uses `wants_bytes()` internally in most places. The next itsdangerous version will do that for the data arg as well, but it's still a prerelease...